docs/tutorials/projects/make-a-plant-watering-robot.md

```diff
@@ -433,8 +399,22 @@ while True:
     if (avg_moisture > 60000):
         print('this plant is too thirsty! giving it more water')
 
-        # Run the water pump for 100 rev. at 1000 rpm
-        await water_pump.go_for(rpm=1000, revolutions=100)
+        # Get the GPIO pin with PWM output (pin number 8) the water pump is
+        # wired to on the board through the relay's IN wire
+        pwm_pin = await local.gpio_pin_by_name(name="8")
+
+        # Run the water pump
+        # Set the duty cycle to .8, meaning that this pin will be in the
+        # high state, powering the motor, for 80% of the duration of the
+        # PWM interval period
+        await pwm_pin.set_pwm(duty=.8)
+
+        # Wait for 15 seconds
+        print('watering')
+        time.sleep(15)
+
+        # Stop the pump by setting the duty cycle to 0%
+        await pwm_pin.set_pwm(duty=.0)
```

Comment:
opt nit: The leading `.` might be missed, especially right next to the `=`. Is this a convention for pwm settings? If not, what do you think of including the leading `0` for the values on lines 410 and 417: `0.8` and `0.0` respectively? Optional.

Reply:
good w that!